src/lib/components/PublishButton/modals/SubmitAndPublishReviewModal.js

```diff
@@ -0,0 +1,163 @@
+                          "acceptAccessToRecord",
+                          data.checked ? "checked" : ""
+                        );
+                      }}
+                      optimized
+                    />
+                    <ErrorLabel
+                      fieldPath="acceptAccessToRecord"
+                      className="mt-0 mb-5"
+                    />
+                  </Form.Field>
+                  <Form.Field>
+                    <RadioField
+                      control={Checkbox}
+                      fieldPath="acceptAfterPublishRecord"
+                      label={
+                        <Trans
+                          defaults="Your upload will be <bold>immediately published</bold> in '{{communityTitle}}'. You will no longer be able to change the files in the upload! However, you will still be able to update the record's metadata later."
+                          values={{
+                            communityTitle: communityTitle,
+                          }}
+                          components={{ bold: <b /> }}
+                        />
+                      }
+                      checked={_get(values, "acceptAfterPublishRecord") === "checked"}
+                      onChange={({ data, formikProps }) => {
+                        formikProps.form.setFieldValue(
+                          "acceptAfterPublishRecord",
+                          data.checked ? "checked" : ""
+                        );
+                      }}
+                      optimized
+                    />
+                    <ErrorLabel
+                      fieldPath="acceptAfterPublishRecord"
+                      className="mt-0 mb-5"
+                    />
+                  </Form.Field>
+                  {publishModalExtraContent && (
+                    <div
+                      dangerouslySetInnerHTML={{ __html: publishModalExtraContent }}
+                    />
+                  )}
+                </Form>
+              </Modal.Content>
+              <Modal.Actions>
+                <Button onClick={onClose} floated="left">
+                  {i18next.t("Cancel")}
+                </Button>
+                <Button
+                  name="submitReview"
+                  onClick={(event) => {
+                    handleSubmit(event);
+                  }}
+                  positive
+                  content={i18next.t("Publish record to community")}
+                />
+              </Modal.Actions>
+            </Modal>
+          );
+        }}
+      </Formik>
+    );
+  }
+}
+
+SubmitAndPublishReviewModal.propTypes = {
+  isConfirmModalOpen: PropTypes.bool.isRequired,
+  community: PropTypes.object.isRequired,
+  onClose: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  initialReviewComment: PropTypes.string,
+  publishModalExtraContent: PropTypes.string,
+};
+
+SubmitAndPublishReviewModal.defaultProps = {
+  initialReviewComment: "",
+  publishModalExtraContent: "",
```

Comment:
question: the component `SubmitAndPublishReviewButtonComponent` set detault props to `undefined`. Why an empty string in this case?

Reply:
Nice catch! I changed it to `undefined` too ;)